README.md

```diff
@@ -1,37 +1,135 @@
- - pip install numpy
- - pip install matplotlib
- 
-## Running dotplots
+If you don't want to install Git (although you should...), you can just save it as `.zip`.
 
-Run the dotplot generator as `./bin/generate_dotplot fastaX fastaY outdotplot`
+## 2. Installing C compiler, Python and dependencies
 
-This will generate a dotplot file which you can plot using `python ./bin/dotplot.py outdotplot`
+In order to use this program, you need to install:
 
-And a `.png` file will be created on the folder
+- C compiler
+- Python 3.x
+- numpy and matplotlib Python packages
 
-## Creating word dictionaries
+## 3. C compiler
 
-Run `bin/dictionary fastas/hemoglobine.fasta output/dictionary`
+MinGW compiler is recommended. It can be downloaded from:
 
-A dictionary will be created on folder `output` with name `dictionary`
+[http://mingw-w64.org/doku.php/download](http://mingw-w64.org/doku.php/download).
 
-## Sorting word dictionaries
+To check the installation, run:
 
-Run `bin/sort_words inputDictionary outputSortedDictionary 6`
+```bash
+gcc --version
+```
 
-(The `inputDictionary` must be created with the previous command)
+The output should look like this:
 
+```bash
+gcc (MinGW.org GCC Build-20200227-1) 9.2.0
+Copyright (C) 2019 Free Software Foundation, Inc.
+This is free software; see the source for copying conditions.  There is NO
+warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+```
 
+## 4. Installing Python 3.x
+
+Python can be downloaded from:
+
+[https://www.python.org/downloads/](https://www.python.org/downloads/).
+
+**Attention! Make sure to check** `Add Python 3.x to PATH` when installing. Otherwise, you will have to do it manually.
+
+![python installer](./assets/tutorial-1.png)
+
+To check the installation, run
+
+```bash
+python --version
+
+pip --version
+```
+
+## 5. Python packages
+
+[`dotplot.py`](./src/dotplot.py) uses [`numpy`](https://numpy.org/) and [`matplotlib`](https://matplotlib.org/).
+
+You can install both with
+
+```bash
+pip install numpy
+pip isntall matplotlib
+```
+
+## 6. Compiling C code
+
+In order to compile the C code, just execute
+
+- `make all` (in Linux)
+- [`make-windows.bat`](make-windows.bat) (in Windows)
+
+The binaries will be compiled in [`./bin/`](./bin).
+
+## 7. Running the generators
+
+### 7.1 generate_dotplot
+
+Execute
+
+```bash
+generate_dotplot <s1> <s2> <output>
+```
+
+where:
+
+- `s1`: first `.fasta` sequence
+- `s2`: second `.fasta` sequence
+- `output`: the output file (plain text)
+
+### 7.2 generate_dotplot_word_thresh
+
+Execute
+
+```bash
+generate_dotplot <s1> <s2> <W> <T> <output>
+```
+
+where:
+
+- `s1`: first `.fasta` sequence
+- `s2`: second `.fasta` sequence
+- `W`: search window size
+- `T`: threshold
+- `output`: the output file (plain text)
+
+## 8. Generating images with Python
+
+The final step is generating the image dotplot from the plain file generated with C, to use it, execute:
+
+```bash
+python .\src\dotplot.py <generated file>
```

Comment:
These slashes are backwards (windows-like), please change them to `/` which is more consistent